pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -583,6 +584,11 @@ func parseAppInstanceConfig(config *zconfig.EdgeDevConfig,
 		appInstance.UUIDandVersion.Version = cfgApp.Uuidandversion.Version
 		appInstance.DisplayName = cfgApp.Displayname
 		appInstance.Activate = cfgApp.Activate
+		if i+1 == lenApps {
+			appInstance.LastDelayedVm = true
+		} else {
+			appInstance.LastDelayedVm = false
+		}
```

Comment:
My basic issue is with the per-device window, and if we remove that this issue goes away.

But even with a per-device window I don't think this would work. pubsub does not ensure ordering between a publisher and subscriber. Publisher can publish UUID 1,2,3 and the subscriber's handler might be called in a different order. All asynchronous programming.

Reply:
Yes, good point.
I've tried to reproduce the state of "wrong order" but it never happened.
But, sure it's not a shred of strong evidence. 
Anycase, if we rethink the design and remove the delay window, this should go away, as I understand.